datadog_checks_dev/datadog_checks/dev/tooling/commands/env/start.py

```diff
@@ -129,15 +129,24 @@ def start(ctx, check, env, agent, python, dev, base, env_vars, org_name, profile
 
     env_type = metadata['env_type']
 
-    agent_ver = agent or os.getenv('DDEV_E2E_AGENT', '6')
+    # TODO: remove this legacy fallback lookup in any future major version bump
+    legacy_fallback = ctx.obj.get('agent', '')
+    if os.path.isdir(legacy_fallback):
```

Comment:
Note that it doesn't work as expected when you have a `~path/dir`, as you need to expand manually.